src/pytorch_lightning/overrides/distributed.py

```diff
@@ -45,8 +45,6 @@ def _find_tensors(
 # https://github.com/pytorch/pytorch/blob/v1.7.1/torch/nn/parallel/distributed.py#L626-L638
 def prepare_for_backward(model: DistributedDataParallel, output: Any) -> None:
     # `prepare_for_backward` is `DistributedDataParallel` specific.
-    if not isinstance(model, DistributedDataParallel):
-        return
```

Comment:
Should this be an assert or raise an exception? Is there any case where this would be allowed?

Reply:
We deleted/moved it because I proposed to move it out. We can make sure in the strategy that this function only gets called on a DDP wrapper.